### PR TITLE
aspnetcore test automation

### PR DESCRIPTION
**Pull request by nathana1**
Adds plaintext and json tests for asp.net core with several variations (Windows/Linux, Mvc, WebListener/Kestrel).

cc:  @sajayantony @damianedwards @halter73

### CODE REVIEW

**Comment by DamianEdwards** [comment]
This doesn't seem to include the actual code, rather it just clones from https://github.com/aspnet/benchmarks as part of the automation script. Don't we need to include the benchmark apps themselves (like all the frameworks here)?

File: frameworks/CSharp/aspnetcore/run-linux.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+fw_depends mono
+sudo apt-get install unzip libunwind8 -y
+
+if [ -d test ]; then
+  rm -rf test
+fi
+
+mkdir test
+cd test
+git clone https://github.com/aspnet/benchmarks.git
```
Comment (msmith-techempower) [inline frameworks/CSharp/aspnetcore/run-linux.sh]: Like @DamianEdwards said, this should be the actual code that handles requests for the various benchmark endpoints and live in our repo.

**Comment by nathana1** [comment]
I'll get something sent out on Monday.

**Comment by nathana1** [comment]
hi @msmith-techempower, just curious if you have any info yet on the timing or the hardware for round 14.  For lots of cores we needed to make tweaks around processor groups and numa to utilize all of the cores so we'd like to make sure we're getting full utilization for your runs as well.  

Thanks.

**Comment by msmith-techempower** [comment]
We are working on the hardware situation at the moment; the timing for round 14 is unknown at the moment.

We expect to run the suite on a few environments: Azure (with minimal cores) and hardware with 40 cores.

**Comment by msmith-techempower** [comment]
This looks good to merge in to me, but there is some discussion about getting this in for round 13.

**Comment by DamianEdwards** [comment]
@msmith-techempower thanks. Do let us know if there's anything we can do to help.

**Comment by msmith-techempower** [comment]
@DamianEdwards We are going to get this merged in for round 13's preview run.

**Comment by DamianEdwards** [comment]
Fantastic, looking forward to it!

**Comment by madhugilla** [comment]
Would having ConfigureAwait as false help in increasing the perf ?

  return context.HttpContext.Response.Body.WriteAsync(_helloWorldPayload, 0, _helloWorldPayload.Length);

**Comment by halter73** [comment]
@madhugilla ASP.NET Core doesn't create a SynchronizationContext , so there typically isn't any performance benefit to using `ConfigureAwait(false)`.

Additionally `return context.HttpContext.Response.Body.WriteAsync(_helloWorldPayload, 0, _helloWorldPayload.Length);` isn't being directly awaited. It's returning a `Task` from a method while `ConfigureAwait()` returns a `ConfiguredTaskAwaitable` which isn't a Task.

**Comment by madhugilla** [comment]
@halter73  lesson learnt, thank you.

**Pull request merged**